kolibri/core/auth/models.py

```diff
@@ -344,6 +344,21 @@ def infer_dataset(self, *args, **kwargs):
         )
 
 
+def validate_username(value):
+    try:
+        validators.EmailValidator()(value)
+    except ValidationError:
+        # for kolibri backwards compatibility, if the username is not an email:
+        validators.RegexValidator(
+            r'[\s`~!@#$%^&*()\-+={}\[\]\|\\\/:;"\'<>,\.\?]',
+            "Enter a valid username. This value can either be an email or contain only letters, numbers, and underscores.",
+            inverse_match=True,
+        )(value)
+        validators.MaxLengthValidator(
```

Comment:
This can be defined in the module scope and reused.

Reply:
Done